src/modules/site-v2/base/views/strains.py

```diff
@@ -143,59 +146,144 @@ def strains_catalog():
+      
+      return redirect(url_for("request_strains.order_confirmation", invoice_hash=order_obj['invoice_hash']), code=302)
+    else:
+      # handle form validation errors
+      title = "Order Summary"
+      cartItems = users_cart['items'] 
+      for item in cartItems:
+        item_price = users_cart.get_price(item)
+        item['price'] = item_price
+      totalPrice = sum(item['price'] for item in cartItems)
+      return render_template('order/order.html', **locals())
 
-  # Fetch items
-  items = form.items.data
   
-  disable_parent_breadcrumb = True
-  title = "Order"
-
-  if (len(items) == 0):
-    flash("You must select strains/sets from the catalog", 'error')
-    return redirect(url_for("request_strains.strains_catalog"))
-
-  # Is the user coming from the catalog?
-  user_from_catalog = request.form.get('from_catalog') == "true"
-
-  # When the user confirms their order it is processed below.
-  if user_from_catalog is False and form.validate_on_submit():
-    order_obj = {'total': form.total,
-                  'date': datetime.now(timezone.utc).date().isoformat(),
-                  'is_donation': False}
-    order_obj.update(form.data)
-    order_obj['phone'] = order_obj['phone'].strip("+")
-    order_obj['items'] = '\n'.join(sorted([u"{}:{}".format(k, v) for k, v in form.item_price()]))
-    order_obj['invoice_hash'] = str(uuid.uuid4()).split("-")[0]
-    order_obj["order_confirmation_link"] = url_for('request_strains.order_confirmation', invoice_hash=order_obj['invoice_hash'], _external=True)
-    send_email({"from": "no-reply@elegansvariation.org",
-                "to": [order_obj["email"]],
-                "cc": config.get("CC_EMAILS"),
-                "subject": "CeNDR Order #" + str(order_obj["invoice_hash"]),
-                "text": ORDER_SUBMISSION_EMAIL_TEMPLATE.format(**order_obj)})
-
-    # Save to google sheet
-    add_to_order_ws(order_obj)
-    flash("Thank you for submitting your order! Please follow the instructions below to complete your order.", 'success')
-    return redirect(url_for("request_strains.order_confirmation", invoice_hash=order_obj['invoice_hash']), code=302)
-  return render_template('order/order.html', **locals())
+  else:
+    """ adding items to the cart """
+    added_items = request.get_json()['strains']
+    if (len(added_items) == 0):
+      flash("You must select strains/sets from the catalog", 'error')
+      return redirect(url_for("request_strains.strains_catalog"))
+ 
+    for item in added_items:
+      users_cart.add_item(item)
+    users_cart.save()
+
+    resp = make_response(jsonify({'status': 'OK'}))
+    if not user:
+      resp.set_cookie('cartID', users_cart.name)
+    
+    return resp
+
+@strains_bp.route('/checkout', methods=['PUT'])
+@jwt_required(optional=True)
+def order_page_remove():
+  """ This view handles removing items from the cart """
+  form = OrderForm()
+  user = get_current_user()
+  item_to_remove = request.get_json()['itemToRemove']

```

Comment:
be careful when doing key look-ups in python dicts, if a key doesn't exist, it will throw a `KeyError`. An alternative that is usually recommended is to use `dict.get('itemToRemove', 'defaultValueIfKeyDoesntExist')`. By default when using .get('key') and not passing a default value, if the key doesn't exist, it will return `None` (which is safer) than a `KeyError`